app/frontend/controllers/notes_controller.js

```diff
@@ -56,16 +58,36 @@ class NotesController {
     this.currentEditor = null;
   }
 
-  initForTab(tabIndex=undefined) {
+  initForTab(tabIndex = undefined) {
     this.reset();
     var currentVerseListFrame = app_controller.getCurrentVerseListFrame(tabIndex);
     if (currentVerseListFrame == null || currentVerseListFrame.length == 0) {
       return;
     }
 
-    var notes = currentVerseListFrame[0].querySelectorAll('.verse-notes');
-    for (var i = 0; i < notes.length; i++) {
-      notes[i].addEventListener('mousedown', (event) => {
+    currentVerseListFrame[0].querySelectorAll('.verse-box').forEach(verseBox => {
+      const verseNotes = verseBox.querySelector('.verse-notes');
+      verseNotes.classList.remove('visible');
+
+      verseBox.querySelector('.notes-info').addEventListener('mousedown', (e) => {
```

Comment:
I'd suggest to move the complete event handler code into a separate function. This would then be similar as handleNotesClick.